pravega/pravega
# Issue 3583: Fix TLS configuration for distributed deployments

**Change log description**  
This PR fixes the issues with TLS setup of the Controller server. It adds additional items to the Controller configuration and uses them in the Controller entry point. 

**Purpose of the change**  
Fixes #3583.

**What the code does**  
* Adds missing configuration items for Zookeeper client and REST server
* Uses these configuration items in Controller entry point

**How to verify it**  
Here are the steps to verify (with Zookeeper setup without TLS): 
* Run a Pravega cluster in distributed mode with TLS and auth enabled. 
* Set the following TLS and Auth parameters for the Controller(s):
  | Configuration parameter | Value | Example/Value used for testing|
  |:---------------------------:|:------|:----------|
  | `controller.auth.tlsEnabled` | `true` | -|
  | `controller.auth.tlsCertFile`  | The server certificate you want to use | `/path/to/pravega/conf/cert.pem`|
  | `controller.auth.tlsTrustStore` | The trusted CA certificate | `/path/to/pravega/conf/cert.pem`  |
 | `controller.auth.tlsKeyFile` | The server's private key file |  `/path/to/pravega/conf/key.pem`  |
 | `controller.zk.secureConnection` | `false` | Setting up secure connection for Zookeeper (ZK) requires setting up its cluster with encryption on. I've not tried that yet, but it should work. |
 | `controller.zk.tlsTrustStoreFile` | The truststore .jks file for ZK |  `/path/to/pravega/conf/standalone.truststore.jks` |
 | `controller.zk.tlsTrustStorePasswordFile` | The file containing the password of the ZK truststore | `/path/to/pravega/conf/standalone.truststore.jks`  |
 | `controller.rest.tlsKeyStoreFile` | The keystore file for REST server to use | `/path/to/pravega/conf/standalone.keystore.jks` | 
| `controller.rest.tlsKeyStorePasswordFile` | The file containing the password of the keystore file used by the REST server | `/path/to/pravega/conf/standalone.keystore.jks.passwd` |
|controller.auth.enabled| `true`| - |
|controller.auth.userPasswordFile | The file containing credentials and ACLs to be used by the default AuthHandler `PasswordAuthHandler` | ``/path/to/pravega/conf/passwd`|
|controller.auth.tokenSigningKey| a string of your choice | I used a super-secret `secret` :)|
